io/net/irqbalance.py

```diff
@@ -0,0 +1,451 @@
+                                             shell=True,
+                                             ignore_status=True
+                                             ).stdout.decode().strip()
+        return self.device_interrupts
+
+    def get_irq_numbers(self):
+        '''
+        Function to get all IRQ numbers assocaited for given device.
+        '''
+        self.irq_number = [int(x.strip(":"))
+                           for x in re.findall(r'\b(\d+):',
+                                               self.device_interrupts)]
+        return self.irq_number
+
+    def get_ping_process_pid(self):
+        """
+        Funtion to get the process ID of ping flood.
+
+        :returns : Process PID number that initated by ping flood command.
+        :rtype : int
+        """
+        cmd = (
+            f"ps -ef | grep '[p]ing -I "
+            f"{self.interface} {self.peer_ip} -c {self.ping_count} -f' "
+            f"| awk '{{print $2}}' | head -1"
+        )
+        process_pid = process.system_output(cmd, shell=True,
+                                            ignore_status=True,
+                                            sudo=True).decode("utf-8")
+        if not process_pid:
+            self.log.debug(f"No more process PID avaialable")
+            return False
+        return process_pid
+
+    def compare_range_strings(self, range_str1, range_str2):
+        '''
+        Function to compare and match cpu range of string format.
+        :rtype : boolean value.
+        '''
+        range_pattern = r'(\d+)-(\d+)'
+        match1 = re.match(range_pattern, range_str1)
+        match2 = re.match(range_pattern, range_str2)
+
+        if match1 and match2:
+            start1, end1 = int(match1.group(1)), int(match1.group(2))
+            start2, end2 = int(match2.group(1)), int(match2.group(2))
+
+            return start1 == start2 and end1 == end2
+        return False
+
+    def cpu_range_validation(self):
+        '''
+        Funtion to validate the assinged CPU's by script.
+        '''
+        self.irq_affinity = '-'.join([str(self.cpu_range[0]),
+                                      str(self.cpu_range[-1])])
+        cmd = f'/proc/irq/{self.irq_number}/smp_affinity_list'
+        self.system_affinity = genio.read_file(cmd)
+        time.sleep(25)
+        if not self.compare_range_strings(self.irq_affinity,
+                                          self.system_affinity):
+            self.fail(f'The smp_affinity_list does not matches to affinity'
+                      ' that was set by script, Please check the logs')
+
+    def grep_module_interrupts(self):
+        '''
+        Funtion to filter all interrupts along associated CPU's of device.
+        '''
+        cmd = f'head -n 1 /proc/interrupts &&' \
+              f' grep {self.interface_type} /proc/interrupts'
+
+        process.run(cmd, shell=True, ignore_status=True
+                    ).stdout.decode().strip()
```

Comment:
I feel you need to return something from this method?
if not,
what are we achieving after grepping something here?
even the output is not captured into some variable which might be used at later part of the code

Reply:
@Naresh-ibm The idea here to just log the "cat /proc/interrupts" for specific device for 5 loop to ensure the interrupts increasing/changing.

we still not have supported code to validate the actual interrupt numbers / IRQ and /CPU since its 3 dimensions huge data set, hence for now we are just capturing the output of "/proc/interrupts" to log file.

We are about open a separate "Todo" item to have 3 dimension irq interrupts validation for /IRQ with /CPU, so this will fulfill our actual validation requirement.